src/stim/py/compiled_measurement_sampler.pybind.cc

```diff
@@ -66,10 +66,22 @@ pybind11::class_<CompiledMeasurementSampler> stim_pybind::pybind_compiled_measur
 }
 
 CompiledMeasurementSampler stim_pybind::py_init_compiled_sampler(
-    const Circuit &circuit, bool skip_reference_sample, const pybind11::object &seed) {
-    simd_bits<MAX_BITWORD_WIDTH> ref_sample = skip_reference_sample
-                                                  ? simd_bits<MAX_BITWORD_WIDTH>(circuit.count_measurements())
-                                                  : TableauSimulator::reference_sample_circuit(circuit);
+    const Circuit &circuit,
+    bool skip_reference_sample,
+    const pybind11::object &seed,
+    const pybind11::object &reference_sample) {
+    uint64_t num_bits = circuit.count_measurements();
+    simd_bits<MAX_BITWORD_WIDTH> ref_sample(circuit.count_measurements());
```

Comment:
Can you avoid unnecessarily initialization and allocating a reference sample that will be replaced in most cases? This thing might be megabytes big; it's a large cost to pay to unconditionally initialize it like this.

Reply:
I moved the initialization / returns into the if/else if that works?

Reply:
Yeah that should work.